ui/selectors/selectors.js

```diff
@@ -1459,3 +1461,29 @@ export function getIsDesktopEnabled(state) {
   return state.metamask.desktopEnabled;
 }
 ///: END:ONLY_INCLUDE_IN
+
+///: BEGIN:ONLY_INCLUDE_IN(flask)
+/**
+ * To get all installed snaps with proper metadata
+ *
+ * @param {*} state
+ * @param {*} onClick - Callback function for handling snap info route on click.
+ * @returns Boolean
+ */
+export function getSnapsList(state, onClick) {
+  const snaps = getSnaps(state);
+  return Object.entries(snaps).map(([key, snap]) => {
+    const targetSubjectMetadata = getTargetSubjectMetadata(state, snap?.id);
+
+    return {
+      key,
+      id: snap.id,
+      packageName: removeSnapIdPrefix(snap.id),
+      name: getSnapName(snap.id, targetSubjectMetadata),
+      onClick: () => {
+        onClick(snap);
```

Comment:
I don't think the `onClick` is necessary here.

Reply:
Removed.